### DISCUSSION TITLE
Retrieving session with supabase-js in Edge Function

### DISCUSSION THREAD

**Discussion by braydenbabbitt** [supabase/supabase#13443] [Questions]
I'm attempting to retrieve the session in an Edge Function in order to get access to the `provider_token` to integrate with a third-party API (Spotify) that I'm using in OAuth for my project.

I've been successful in retrieving the session and token in the browser, but when using the `supabaseClient.auth.getSession()` in my Edge Function, no session is returned.

I'm basically following this example: https://supabase.com/docs/guides/functions/auth but replacing `await supabaseClient.auth.getUser()` with `await supabaseClient.auth.getSession()`, but the data returned is always `null`.

Am I missing something? I can provide any other info needed.

![supabase-edge-function-session-issue](https://user-images.githubusercontent.com/1495342/229979064-95206ea9-647e-4f62-9325-0b2fae38b962.png)

**Comment by GaryAustin1** [comment]
There is no session set up when you set the authorization header only.   That just allows you make calls with the user jwt (access token).    getUser goes to the database to load your user information with the jwt.  getSession tries to load the session from the client which has no session in your case.   getSession also does not work if there is no local storage and persistSession:false is not set in createClient.
